app/models/dataset/value_set.rb

```diff
@@ -20,7 +20,7 @@ class Dataset::ValueSet < ApplicationRecord
   has_many :values, foreign_key: 'dataset_value_set_id', inverse_of: :value_set
 
   accepts_nested_attributes_for :values, reject_if: proc { |attributes|
-    attributes['dataset_key_id'].blank?
+    attributes['dataset_key_id'].blank? || attributes['value'].blank?
```

Comment:
This results in us not saving a `Dataset::Value` for a key if it was empty.

For example, if we have 3 fields to fill in, but I only fill in 2, we only get 2 `Dataset::Value` records.

![Screenshot 2020-05-22 at 13 20 21](https://user-images.githubusercontent.com/282788/82667126-1ea8db00-9c2f-11ea-924f-b7046bdca45a.png)

This means we'll have to handle missing values in the exporter.

I'm not sure on the right thing to do here.

On one hand I think that if the submission is successful, we should store a `Dataset::Value` for every field (because it _was_ submitted, just with no content).

On the other hand, you _can_ just infer that through the lack of persisted `Dataset::Value`.

Reply:
How the changes in f59590ba2 ?